asteroid/masknn/blocks.py

```diff
@@ -40,12 +44,16 @@ def __init__(self, in_chan, hid_chan, skip_out_chan, kernel_size, padding,
                                           conv_norm(hid_chan), depth_conv1d,
                                           nn.PReLU(), conv_norm(hid_chan))
         self.res_conv = nn.Conv1d(hid_chan, in_chan, 1)
-        self.skip_conv = nn.Conv1d(hid_chan, skip_out_chan, 1)
+        if skip_out_chan:
+            self.skip_conv = nn.Conv1d(hid_chan, skip_out_chan, 1)
 
     def forward(self, x):
         """ Input shape [batch, feats, seq]"""
         shared_out = self.shared_block(x)
         res_out = self.res_conv(shared_out)
+        if not self.skip_out_chan:
+            # Return it twice to have consistent number of outputs
+            return res_out, res_out
```

Comment:
@popcornell Only thing I'm unsure about : whether I should have consistent outputs or rather output only res_out..

Reply:
No I think is better to return only one argument, in that way is easier to catch errors in the implementation.

Reply:
You're right, changing it